soufi/tests/finders/test_yum_finder.py

```diff
@@ -338,7 +321,83 @@ def test__test_url_timeout(self):
+        lxml.return_value = b"<xml>Test</xml>"
+        compress = self.patch(lzma, 'compress')
+        yum.get_repomd(queue, url)
+        compress.assert_called_once_with(lxml.return_value)
+        queue.put.assert_called_once_with(
+            (str(repo.return_value.baseurl), compress.return_value)
+        )
+
+    def test_get_repomd_http_error(self):
+        url = self.factory.make_url()
+        queue = mock.MagicMock()
+        self.patch(repomd, 'load').side_effect = urllib.error.HTTPError(
+            None, None, None, None, None
+        )
+        lxml = self.patch(repomd.defusedxml.lxml, 'tostring')
+        compress = self.patch(lzma, 'compress')
+        yum.get_repomd(queue, url)
+        lxml.assert_not_called()
+        compress.assert_not_called()
+        queue.put.assert_called_once_with((None, None))
+
+    def test_lookup_in_repomd(self):
+        queue = mock.MagicMock()
+        package = mock.MagicMock()
+        baseurl = self.factory.make_url()
+        repomd_xml = self.factory.make_string()
+        name = self.factory.make_string()
+        decompress = self.patch(lzma, 'decompress')
+        lxml = self.patch(repomd.defusedxml.lxml, 'fromstring')
+        repo = self.patch(repomd, 'Repo')
+        repo.return_value.findall.return_value = [package]
+        yum.lookup_in_repomd(queue, baseurl, repomd_xml, name)
+        decompress.assert_called_once_with(repomd_xml)
+        lxml.assert_called_once_with(decompress.return_value)
+        repo.assert_called_once_with(baseurl, lxml.return_value)
+        queue.put.assert_called_once_with([yum.serialize_package(package)])
+
+    def test_lookup_in_repomd_no_baseurl(self):
+        queue = mock.MagicMock()
+        repomd_xml = self.factory.make_string()
+        name = self.factory.make_string()
+        decompress = self.patch(lzma, 'decompress')
+        repo = self.patch(repomd, 'Repo')
+        yum.lookup_in_repomd(queue, None, repomd_xml, name)
+        decompress.assert_not_called()
+        repo.assert_not_called()
+        queue.put.assert_called_once_with([])
+
+    def test_lookup_in_repomd_no_repoxml(self):
+        queue = mock.MagicMock()
+        baseurl = self.factory.make_url()
+        name = self.factory.make_string()
+        decompress = self.patch(lzma, 'decompress')
+        repo = self.patch(repomd, 'Repo')
+        yum.lookup_in_repomd(queue, baseurl, None, name)
+        decompress.assert_not_called()
+        repo.assert_not_called()
+        queue.put.assert_called_once_with([])
+
+    def test_do_task(self):
+        data = self.factory.make_string('response')
+        queue = self.patch(yum, 'Queue')
+        queue.return_value.get.return_value = data
+        process = self.patch(yum, 'Process')
+        process.return_value.is_alive.return_value = True
+        response = yum.do_task('a', 'b', 'c')
+        self.assertEqual(data, response)
+        process.return_value.terminate.assert_called_once_with()
```

Comment:
I think these tests would read better and look more focused if you refactor the mock set up into a function that returns them. It's quite hard to see what's being tested; the inputs, outputs and assertions need to be very clear. Perhaps add some comments if the long setup is unavoidable so test intent is more obvious. For example, this `test_do_task` doesn't tell me *what* it's testing, just that it's calling it.

Reply:
I suppose I did get kind of spartan with explaining my tests near the end of this.  I'll see what I can do.

Reply:
Much more betterer, thanks.